src/connectors/range/connectRange.js

```diff
@@ -275,9 +241,17 @@ export default function connectRange(renderFn, unmountFn = noop) {
       dispose({ state }) {
         unmountFn();
 
-        return state
-          .removeNumericRefinement(attribute)
-          .removeDisjunctiveFacet(attribute);
+        const stateWithoutDisjunctive = state.removeDisjunctiveFacet(attribute);
+
+        // can not use setQueryParameters || removeNumericRefinements, because
+        // they both keep the old value. This isn't immutable, but it is fine
+        // since it's already a copy.
+        stateWithoutDisjunctive.numericRefinements = {
```

Comment:
I'm not sure if we can find a better solution for this

Reply:
IIRC you can use `removeNumericRefinements` with the attribute only, it will perform a clear under the hood.

Reply:
It doesn’t actually remove the refinement, I havent found a solution which doesn’t break for refining to an empty state